crates/services/parallel-executor/src/scheduler/contracts_changes.rs

```diff
@@ -0,0 +1,59 @@
+use fuel_core_storage::transactional::Changes;
+use fuel_core_types::fuel_tx::ContractId;
+use fxhash::FxHashMap;
+
+#[derive(Debug, Clone, Default)]
+pub struct ContractsChanges {
+    contracts_changes: FxHashMap<ContractId, u64>,
+    latest_id: u64,
+    changes_storage: FxHashMap<u64, (Vec<ContractId>, Changes)>,
+}
+
+impl ContractsChanges {
+    pub fn new() -> Self {
+        Self {
+            contracts_changes: FxHashMap::default(),
+            changes_storage: FxHashMap::default(),
+            latest_id: 0,
+        }
+    }
+
+    pub fn add_changes(&mut self, contract_ids: &[ContractId], changes: Changes) {
+        let id = self.latest_id;
```

Comment:
Should this and `latest_id` be `index` instead of `id`? We are incrementing them like indices. It's just confusing having this `id` and the `contract_id` together.